pkg/services/searchV2/auth.go

```diff
@@ -15,28 +18,46 @@ type ResourceFilter func(kind entityKind, uid, parentUID string) bool
 
 // FutureAuthService eventually implemented by the security service
 type FutureAuthService interface {
-	GetDashboardReadFilter(user *user.SignedInUser) (ResourceFilter, error)
+	GetDashboardReadFilter(ctx context.Context, orgID int64, user *user.SignedInUser) (ResourceFilter, error)
 }
 
 var _ FutureAuthService = (*simpleAuthService)(nil)
 
 type simpleAuthService struct {
-	sql db.DB
-	ac  accesscontrol.Service
+	sql           db.DB
+	ac            accesscontrol.Service
+	features      featuremgmt.FeatureToggles
+	folderService folder.Service
+	logger        log.Logger
 }
 
 type dashIdQueryResult struct {
 	UID string `xorm:"uid"`
 }
 
-func (a *simpleAuthService) GetDashboardReadFilter(user *user.SignedInUser) (ResourceFilter, error) {
+func (a *simpleAuthService) GetDashboardReadFilter(ctx context.Context, orgID int64, user *user.SignedInUser) (ResourceFilter, error) {
 	if !a.ac.IsDisabled() {
 		canReadDashboard, canReadFolder := accesscontrol.Checker(user, dashboards.ActionDashboardsRead), accesscontrol.Checker(user, dashboards.ActionFoldersRead)
 		return func(kind entityKind, uid, parent string) bool {
+			var scopes []string
+			if a.features.IsEnabled(featuremgmt.FlagNestedFolders) {
+				var err error
+				if kind == entityKindFolder {
+					scopes, err = dashboards.GetInheritedScopes(ctx, orgID, uid, a.folderService)
+				} else if kind == entityKindDashboard && parent != folder.GeneralFolderUID {
+					scopes, err = dashboards.GetInheritedScopes(ctx, orgID, parent, a.folderService)
+				}
+				if err != nil {
+					a.logger.Warn("could not retrieve inherited folder scopes:", "err", err)
+				}
+			}
```

Comment:
the folder service `GetParents()` (called by `GetInheritedScopes()`) checks if the flag is enabled an if not then returns an empty array:
https://github.com/grafana/grafana/blob/30b4205521bb38f11a037c43f0c3c2059b92579e/pkg/services/folder/folderimpl/folder.go#L184-L189

So in that case no inherited scopes are returned therefore I think that we don't need the check for the flag here.

Reply:
Ah, good point! I'll remove the extra check.